src/Compilers/Core/Portable/Compilation/Compilation.cs

```diff
@@ -2627,9 +2631,10 @@ internal void EnsureAnonymousTypeTemplates(CancellationToken cancellationToken)
             if (getMetadataPeStreamOpt != null)
             {
                 Debug.Assert(!metadataOnly);
+                Debug.Assert(!context.IsRefAssembly);
 
                 if (!Cci.PeWriter.WritePeToStream(
-                    context,
+                    context.WithIsRefAssembly(!includePrivateMembers),
```

Comment:
>context.WithIsRefAssembly(!includePrivateMembers) [](start = 20, length = 49)

Here as well. I guess we are missing a clear definition of what does RefAssembly == true mean.